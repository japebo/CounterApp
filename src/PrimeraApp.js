import React from 'react';
import PropTypes from 'prop-types'

const PrimeraApp = ( { nombre, subtitulo } ) => { //usando desestructuración
    
    return ( //a continuación se utiliza un Fragment para poder devolver más de una lía de código HTML
    <> 
      <h1> Hola mundo, soy {nombre}. </h1>
      {/* <p>Esta es mi primera aplicación en React</p> */}
      <p>{subtitulo}</p>
    </>
    );
};

PrimeraApp.propTypes = {
  nombre: PropTypes.string.isRequired
};

PrimeraApp.defaultProps = {
  subtitulo: 'Soy un subtítulo'
};

export default PrimeraApp;