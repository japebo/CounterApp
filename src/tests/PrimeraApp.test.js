import React from 'react';
// import { render } from "@testing-library/react";
// import '@testing-library/jest-dom'
import PrimeraApp from "../PrimeraApp";
import { shallow } from 'enzyme';

describe('Pruebas en <PrimeraApp />', () => {
    test('Debe mostrar el mensaje "Hola mundo, soy Javier"', () => {
        // const nombre = 'Javier';
        // const { getByText } = render(<PrimeraApp nombre={nombre}/>);
        // expect( getByText('Hola Mundo, soy Javier')).toBeInTheDocument();

        const wrapper = shallow(<PrimeraApp nombre='Javier'/>);

        expect(wrapper).toMatchSnapshot();
    })
    
    test('Debe mostrar el subtítulo enviado por props', () => {
        const nombre = 'Sandra';
        const subtitulo = 'Esto es un subtítulo';

        const wrapper = shallow(<PrimeraApp nombre={nombre} subtitulo={subtitulo}/>);

        const textoParrafo = wrapper.find('p').text(); //NOTA: el metodo find funciona parecido a como lo hace document.querySelector()

        expect( textoParrafo ).toBe(subtitulo);
    })
    
})
