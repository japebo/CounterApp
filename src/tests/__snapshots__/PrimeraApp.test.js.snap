// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe mostrar el mensaje "Hola mundo, soy Javier" 1`] = `
<Fragment>
  <h1>
     Hola mundo, soy 
    Javier
    . 
  </h1>
  <p>
    Soy un subt√≠tulo
  </p>
</Fragment>
`;
